vignettes/Workflow.Rmd

```diff
@@ -0,0 +1,495 @@
+typically want to apply any procedure to all lanes. 
+Here is a simple summary function that
+computes the frequency distribution of the number of reads.
+
+```{r}
+islandReadSummary <- function(x)
+{
+    g <- resize(x, 200)
+    s <- slice(coverage(g), lower = 1)
+    tab <- table(viewSums(s) / 200)
+    df <- DataFrame(tab)
+    colnames(df) <- c("chromosome", "nread", "count")
+    df$nread <- as.integer(df$nread)
+    df
+}
+```
+
+Applying it to our test-case, we get
+
+```{r}
+head(islandReadSummary(cstest$ctcf)) 
+``` 
+
+We can now use it to summarize
+the full dataset, flattening the returned *DataFrameList* with the
+`stack` function. 
+
+```{r} 
+nread.islands <- DataFrameList(lapply(cstest, islandReadSummary)) 
+nread.islands <- stack(nread.islands, "sample") 
+nread.islands 
+```
+
+```{r}
+xyplot(log(count) ~  nread | sample, as.data.frame(nread.islands),
+       subset = (chromosome == "chr10" & nread <= 40), 
+       layout = c(1, 2), pch = 16, type = c("p", "g"))
+```
+
+```{r fig.height=10, echo=FALSE}
+plot(trellis.last.object()) 
+```
+
+If reads were sampled randomly from the genome, then the null
+distribution number of reads per island would have a geometric
+distribution; that is, 
+$$P(X = k) = p^{k-1} (1-p)$$ 
+In other words, $\log P(X = k)$ is linear in $k$. Although our 
+samples are not random, the islands with just one or two reads may be 
+representative of the null distribution. 
+
+```{r}
+xyplot(log(count) ~ nread | sample, data = as.data.frame(nread.islands), 
+       subset = (chromosome == "chr10" & nread <= 40), 
+       layout = c(1, 2), pch = 16, type = c("p", "g"), 
+       panel = function(x, y, ...) {
+           panel.lmline(x[1:2], y[1:2], col = "black")
+           panel.xyplot(x, y, ...)
+       })
+```
+
+```{r fig.height=8}
+plot(trellis.last.object()) 
+```
+
+We can create a similar plot of the distribution of depths. 
+
+```{r}
+islandDepthSummary <- function(x) 
+{
+ g <- resize(x, 200) 
+ s <- slice(coverage(g), lower = 1) 
+ tab <- table(viewMaxs(s) / 200) 
+ df <- DataFrame(tab) 
+ colnames(df) <- c("chromosome", "depth", "count")
+ df$depth <- as.integer(df$depth) 
+ df
+} 
+
+depth.islands <- DataFrameList(lapply(cstest, islandDepthSummary))
+depth.islands <- stack(depth.islands, "sample")
+
+xyplot(log(count) ~ depth | sample, as.data.frame(depth.islands),
+       subset = (chromosome == "chr10" & depth <= 20), 
+       layout = c(1, 2), pch = 16, type = c("p", "g"), 
+       panel = function(x, y, ...){
+           lambda <- 2 * exp(y[2]) / exp(y[1]) 
+           null.est <- function(xx) {
+               xx * log(lambda) - lambda - lgamma(xx + 1)
+           } 
+           log.N.hat <- null.est(1) - y[1]
+           panel.lines(1:10, -log.N.hat + null.est(1:10), col = "black")
+           panel.xyplot(x, y, ...)
+       })
+## depth.islands <- summarizeReads(cstest, summary.fun = islandDepthSummary)
+```
+
+```{r fig.height=10, echo=FALSE}
+plot(trellis.last.object()) 
+```
```

Comment:
Lines 233-265 are the code blocks we'll do differently because we have the comments on line 260 that should stay with the code block. (If you look at them in the knitted file they show up after the plot rather than with the rest of the code.) So we should save the plot to a variable and later 'return' that variable in another code block. Please edit as follows:

On Line 248, set the `xyplot` to a `plt`:

    plt <- xyplot(log(count) ~ depth | sample, as.data.frame(depth.islands),

On line 264, replace `plot(trellis.last.object())` with `plt`.